### PR TITLE
Mark project as dirty when renaming a layer/group

### PR DESCRIPTION
**Pull request by nirvn**
## Description
@nyalldawson , that's pretty much the only action I'm doing regularly which IMHO should trigger a dirtying of the project, all good with you? 

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

- [x] Commit messages are descriptive and explain the rationale for changes
- [x] Commits which fix bugs include `fixes #11111` in the commit message next to the description
- [x] Commits which add new features are tagged with `[FEATURE]` in the commit message
- [x] Commits which change the UI or existing user workflows are tagged with `[needs-docs]` in the commit message and contain sufficient information in the commit message to be documented
- [x] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [x] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [x] New unit tests have been added for core changes
- [x] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTE.md#contributing-to-qgis) before each commit

### CODE REVIEW

**Comment by nirvn** [comment]
@m-kuhn , on the basis of your comment on that other dirty PR, I've updated the dirtying upon layer CRS change in commit 4bfa793. Cheers.

**Comment by nirvn** [comment]
@nyalldawson , I think we've reached an agreement on this, OK to merge?

**Comment by m-kuhn** [comment]
What about the proposal to connect this all to `configChanged` in the `QgsVectorLayer` constructor?

**Comment by nirvn** [comment]
@m-kuhn , CRS and layer renaming are also needed for raster layers, what should we do there?

**Comment by m-kuhn** [comment]
Put the connection in the  QgsMapLayer constructor?

**Comment by nirvn** [comment]
PR updated to take into account comments.

File: src/core/qgsvectorlayer.cpp [hunk 1]
```diff
@@ -896,7 +896,10 @@ bool QgsVectorLayer::setSubsetString( const QString &subset )
   updateFields();
 
   if ( res )
+  {
+    emit configChanged();
```
Comment (m-kuhn) [inline src/core/qgsvectorlayer.cpp]: I'd probably still make a `setSubsetString()`, that's nice to have anyway :)

Comment (nirvn) [reply src/core/qgsvectorlayer.cpp]: @m-kuhn , OK, I was thinking about this too.

Comment (m-kuhn) [reply src/core/qgsvectorlayer.cpp]: And for extra points, add the Q_PROPERTY for subsetString on top

Comment (nyalldawson) [reply src/core/qgsvectorlayer.cpp]: And for triple points, add unit tests with signal spies for all these signals :)

**Comment by nirvn** [comment]
@m-kuhn , signal & Q_PROPERTY added; @nyalldawson , subset string & signal test case added.

File: src/core/qgsvectorlayer.h [hunk 2]
```diff
@@ -2148,6 +2149,10 @@ class CORE_EXPORT QgsVectorLayer : public QgsMapLayer, public QgsExpressionConte
      */
     void updatedFields();
 
+    /**
+     * Emitted when the layer's subset string has changed.
```
Comment (m-kuhn) [inline src/core/qgsvectorlayer.h]: \since

File: tests/src/python/test_qgsvectorlayer.py [hunk 3]
```diff
@@ -1619,6 +1619,18 @@ def testUniqueStringsMatching(self):
...
                          set(['orange', 'BanaNa', 'waterMelon', 'pineapple', 'coconut']))
 
+    def onSubsetStringChanged(self):
+        self.subsetStringChanged = True
+
+    def test_subsetString(self):
+        self.subsetSringChanged = False
+        myPath = os.path.join(unitTestDataPath(), 'lines.shp')
+        layer = QgsVectorLayer(myPath, 'test', 'ogr')
+        layer.subsetStringChanged.connect(self.onSubsetStringChanged)
```
Comment (m-kuhn) [inline tests/src/python/test_qgsvectorlayer.py]: Nitpicking here, but given that there's a typo in the variable names here you'll need to revisit this anyway ;)

you could use a local function

```
def test_subsetString(self):
    subset_string_changed = False
    def onSubsetStringChanged():
            subset_string_changed = True
    ...
```

**Pull request merged**